gnuradio-runtime/lib/CMakeLists.txt

```diff
@@ -68,6 +63,13 @@ GR_INCLUDE_SUBDIRECTORY(thread)
 GR_INCLUDE_SUBDIRECTORY(math)
 GR_INCLUDE_SUBDIRECTORY(controlport)
 
+########################################################################
+# Setup CTRLPORT include path, if using
+########################################################################
+if(ENABLE_CTRLPORT_THRIFT)
+    include_directories(${THRIFT_INCLUDE_DIRS})
```

Comment:
Apparently `${THRIFT_INCLUDE_DIRS}` is not defined on any Linux distro. Maybe it's a naming issue?
That's what the build machines are trying to tell you.

Reply:
Interesting. Let me look into the naming of this variable. Do the build machines succeed with the original code (which also uses this variable name)? That would be quite odd  ...

Reply:
So unless GR's CMakeLists.txt scripts are not using the local cmake/Modules/FindThrift.cmake, then this variable will be used. See, e.g., https://github.com/gnuradio/gnuradio/blob/master/cmake/Modules/FindThrift.cmake#L84 . This usage is OS-independent.

Reply:
So, would it make sense to put it after the ` if(THRIFT_FOUND)` stanza?

Reply:
We're already using `OPTION(ENABLE_CTRLPORT_THRIFT "Enable ControlPort Thrift support" ON)` which will be TRUE if Thrift is found and matches whatever specs are required for it. We don't want to allow any old Thrift, after all :)

Reply:
Reading: https://cmake.org/cmake/help/v3.0/command/option.html We don't match any thrift version before the find_package(thrift) call, or can you enlighten me what I'm missing here. Anyway the build should pass if thrift is not available on a build machine, right?

Reply:
the `ENABLE_CTRLPORT_THRIFT` is a pure user defined variable which is set to ON by default (which then enables the search for thrift). instead of ENABLE_CTRLPORT_THRIFT the right check should be like in the other file `if(FOUND_THRIFT)`

Reply:
The build should pass if Thrift is not found: true! Let me verify locally, but a quick read through the CMake code looks OK from this perspective to me.

Reply:
But OK yeah I get your point about ENABLE_CTRLPORT_THRIFT versus FOUND_THRIFT. Note that there's also ENABLE_GR_CTRLPORT, CTRLPORT_BACKENDS, GR_RPCSERVER_ENABLED, and GR_RPCSERVER_THRIFT ... and maybe more! Holy crap ctrlport's CMake stuff is complicated!

Reply:
And CTRLPORT_BACKENDS too.

Reply:
My recommendation is to leave FOUND_THRIFT alone & not use it except for setting ENABLE_CTRLPORT_THRIFT. Then, change the CMake code in gnuradio-runtime/lib/controlport/CMakeLists.txt such that if FOUND_THRIFT isn't TRUE, then ENABLE_CTRLPORT_THRIFT will be set to false. If there's a good way to figure out if the user selected ENABLE_CTRLPORT_THRIFT to be TRUE, then have CMake error out with a reasonable debug printout -- like what it does with ENABLE_GR_* if it doesn't meet enabling criteria. I think this will take care of the immediate issue you found, @noc0lour .

Reply:
So the issue here is if Thrift is enabled but not found, then the variable ENABLE_CTRLPORT_THRIFT will be set and TRUE, while THRIFT_INCLUDE_DIRS will not be set and/or undefined. Are @noc0lour noted in discussion, we should be checking for both here. I'll fix this check to do both.